open-banking-accelerator/components/com.wso2.openbanking.accelerator.common/src/main/java/com/wso2/openbanking/accelerator/common/util/JWTUtils.java

```diff
@@ -120,4 +137,105 @@ public static boolean validateJWTSignature(String jwtString, String jwksUri, Str
         return true;
     }
 
+    /**
+     * Validates the signature of a given JWT against a given public key.
+     *
+     * @param signedJWT the signed JWT to be validated
+     * @param publicKey the public key that is used for validation
+     * @return true if signature is valid else false
+     * @throws NoSuchAlgorithmException if the given algorithm doesn't exist
+     * @throws InvalidKeySpecException if the provided key is invalid
+     * @throws JOSEException if an error occurs during the signature validation process
+     */
+    @Generated(message = "Excluding from code coverage as KeyFactory does not initialize")
+    public static boolean validateJWTSignature(SignedJWT signedJWT, String publicKey)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, JOSEException, OpenBankingException {
+
+        byte[] publicKeyData = Base64.getDecoder().decode(publicKey);
+        X509EncodedKeySpec spec = new X509EncodedKeySpec(publicKeyData);
+        // Example : RS256
+        String algorithm = signedJWT.getHeader().getAlgorithm().getName();
+        KeyFactory kf;
+        // In here if the algorithm is directly passes (like RS256) it will generate exceptions
+        // hence RSA should be passed
+        if (algorithm.indexOf(RS) == 0) {
+            kf = KeyFactory.getInstance(ALGORITHM_RSA);
+        } else {
+            throw new OpenBankingException("Algorithm " + algorithm + " not yet supported.");
+        }
```

Comment:
Shall we move this to a private method and return the KeyFactory instance from it?

Reply:
Resolved in [83e631f](https://github.com/wso2/financial-open-banking/pull/67/commits/83e631f5cf7f0e52f6460082da104e93ca98e07e).